_data-prepper/pipelines/configuration/processors/translate.md

```diff
@@ -0,0 +1,171 @@
+
+
+## Configuration
+
+You can use the following options to configure the `translate` processor.
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| mappings | No | List | Define inline mappings. See [mappings](#mappings) for details |
+| file | No | Map | Define the file that contains mapping configurations. See [file](#file) for details |
+
+### mappings
+
+Each item in `mappings` has the following options:
+
+| Parameter | Required | Type | Description |
+| :--- | :--- | :--- | :--- |
+| source | Yes | String or List | The source field to translate. Can be a string or a list of strings. |
+| targets | Yes | List | The target fields configurations, include target field key, translation maps, etc. |
+
+Each item in `targets` list has the following options:
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| target | Yes | String | The key to specify the field in the output where the translated value should be placed |
+| map | No | Map | The map consists of key-value pairs that define the translations. Each key represents a possible value in the source field, and the corresponding value represents what it should be translated to. See [map option](#map-option) for examples. At least one of `map` and `regex` should be configured. |
+| regex | No | Map | Use regex patterns as keys to define the translation map. See [regex option](#regex-option) for more details. At least one of `map` and `regex` should be configured. |
+| default | No | String | Use this default value when no match is found during translation |
+| type | No | String | Specify the type of the data for the target value |
+| translate_when | No | String | Use a [Data Prepper expression]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/) to specify a condition for performing the translation. If specified, translate only when the expression evaluates to true. |
+
+#### map option
+The keys provided in the map option can be of the following types:
+* Individual keys
+  ```yaml
+    map:
+      ok : "Success"
+      120: "Found"
+  ```
+* Number ranges
+  ```yaml
+    map:
+      "100-200": "Success"
+      "400-499": "Error"
+  ```
+* Comma-delimited keys
+  ```yaml
+    map:
+      "key1,key2,key3": "value1"
+      "100-200,key4": "value2"
+  ```
+
+  While configuring the keys in map, there cannot be any overlapping number ranges or duplicate keys.
+
+#### regex option
+`regex` can be specified with the following options.
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| patterns | Yes | Map | Key-value pairs defining the regex patterns of keys and the value to translate to for each pattern |
+| exact | No | Boolean | Whether to use full string match or partial string match on the regex patterns. If true, it is considered a match only when the entire key matches the pattern. Otherwise, it is considered a match as long as a substring of the key matches the pattern. |
```

Comment:
```suggestion
| exact | No | Boolean | Whether to use full string match or partial string match on the regex patterns. If `true`, the pattern is considered a match only when the entire key matches the pattern. Otherwise, the pattern is considered a match as long as a sub-string of the key matches the pattern. |
```